ansible/ocs.yaml

```diff
@@ -0,0 +1,292 @@
+
+    - name: Wait for storage baremetal hosts to become ready
+      shell: oc get bmh -l test-storage=yes -n openshift-machine-api
+      retries: 100
+      delay: 30
+      register: storage_bmhs_ready
+      until: (storage_bmhs_ready.stdout | regex_findall('OK       ready') | length) == 3 or (storage_bmhs_ready.stdout | regex_findall('OK       provisioned') | length) == 3
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Scale storage machineset to 3
+      shell: oc scale machineset/{{ ocp_cluster_name }}-storage-0 --replicas=3 -n openshift-machine-api
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait for storage nodes to be provisioned
+      shell: oc get nodes -l node-role.kubernetes.io/storage
+      retries: 100
+      delay: 30
+      register: storage_nodes_ready
+      until: (storage_nodes_ready.stdout | regex_findall('Ready    storage') | length) == 3
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Label storage nodes as local-storage capable
+      shell: |
+        for i in $(oc get nodes --no-headers -l node-role.kubernetes.io/storage -o name); do
+          oc label $i cluster.ocs.openshift.io/openshift-storage=''
+        done
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+    
+    - name: Deploy local-storage subscription
+      shell: |
+        oc new-project local-storage
+        oc annotate project local-storage --overwrite openshift.io/node-selector=''
+        oc apply -f {{ working_yamls_dir }}/ocs/local-storage-sub.yaml
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait for local-storage operator to be ready
+      shell: oc get pods -n local-storage
+      retries: 100
+      delay: 20
+      register: local_storage_operator_ready
+      until: (local_storage_operator_ready.stdout | regex_findall('local-storage-operator-.+-.+Running') | length) == 1
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Create local-storage PVs
+      shell: |
+        oc apply -f {{ working_yamls_dir }}/ocs/local-storage-volumes.yaml
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
+
+    - name: Wait for local-storage PVs to be ready
+      shell: oc get pv -n local-storage
+      retries: 100
+      delay: 20
+      register: local_storage_volumes_ready
+      until: (local_storage_volumes_ready.stdout | regex_findall('local-pv-.+Available') | length) == (local_storage_volumes_ready.stdout | regex_findall('local-pv-.+') | length) and (local_storage_volumes_ready.stdout | regex_findall('local-pv-.+Available') | length) != 0
+      environment:
+        PATH: "{{ oc_env_path }}"
+        KUBECONFIG: "{{ kubeconfig }}"
```

Comment:
Do you actually have to wait here? Will the OCS deployment not just pause until it becomes schedulable?

Reply:
That's a fair point.